src/economy/economy.cpp

```diff
@@ -1398,7 +1398,8 @@ void populate_private_construction_consumption(sys::state& state) {
 
 	for(auto c : state.world.in_province_building_construction) {
 		auto owner = c.get_nation().id;
-		if(owner && owner == c.get_province().get_nation_from_province_control() && c.get_is_pop_project() && province_building_type(c.get_type()) == economy::province_building_type::railroad) {
+		// Rationale for not checking building type: Its an invalid state; should not occur under normal circumstances
+		if(owner && owner == c.get_province().get_nation_from_province_control() && c.get_is_pop_project()) {
```

Comment:
we check building type because both public and private province building constructions can end up in this relationship. But only the privately built railroads are tallied up here

Reply:
ah I see, you are referring to not checking for being a railroad